File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Adresse.java
package no.nav.melosys.eessi.controller.dto;

import lombok.Data;
import no.nav.melosys.eessi.service.sed.helpers.LandkodeMapper;
import org.apache.commons.lang3.StringUtils;

@Data
public class Adresse {

    private String poststed;
    private String postnr;
    private String land;
    private String gateadresse;
    private String tilleggsnavn;
    private String region;
    private Adressetype adressetype;

    public static Adresse av(no.nav.melosys.eessi.models.sed.nav.Adresse adresseFraRina) {
        if (adresseFraRina == null) {
            return new Adresse();
        }

        Adresse adresse = new Adresse();

        adresse.poststed = adresseFraRina.getBy();
        adresse.postnr = adresseFraRina.getPostnummer();
        adresse.land = LandkodeMapper.mapTilNavLandkode(adresseFraRina.getLand());
        adresse.gateadresse = String.format("%s %s",
            StringUtils.defaultIfEmpty(adresseFraRina.getGate(), ""),
            StringUtils.defaultIfEmpty(adresseFraRina.getBygning(), "")
        ).trim();
        adresse.tilleggsnavn = StringUtils.defaultIfEmpty(adresseFraRina.getBygning(), "");
        adresse.region = adresseFraRina.getRegion();
        adresse.adressetype = Adressetype.fraAdressetypeRina(adresseFraRina.getType());

        return adresse;
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Adresse.kt
package no.nav.melosys.eessi.controller.dto

import no.nav.melosys.eessi.models.sed.nav.Adresse as RinaAdresse
import no.nav.melosys.eessi.service.sed.helpers.LandkodeMapper
import org.apache.commons.lang3.StringUtils

data class Adresse(
    var poststed: String? = null,
    var postnr: String? = null,
    var land: String? = null, // kaster NullPointerException i java kode, men av må kunne lage object med null-verdier
    var gateadresse: String? = null,
    var tilleggsnavn: String? = null,
    var region: String? = null,
    var adressetype: Adressetype? = null
) {
    companion object {
        fun av(adresseFraRina: RinaAdresse?): Adresse = adresseFraRina?.let {
            Adresse(
                poststed = it.by,
                postnr = it.postnummer,
                land = LandkodeMapper.mapTilNavLandkode(it.land),
                gateadresse = "${StringUtils.defaultIfEmpty(it.gate, "")} ${StringUtils.defaultIfEmpty(it.bygning, "")}".trim(),
                tilleggsnavn = StringUtils.defaultIfEmpty(it.bygning, ""),
                region = it.region,
                adressetype = Adressetype.fraAdressetypeRina(it.type)
            )
        } ?: Adresse()
    }
}
